src/components/FlatParameters/FlatParameters.js

```diff
@@ -0,0 +1,291 @@
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import React, { useCallback, useState } from 'react';
+import makeStyles from '@mui/styles/makeStyles';
+import {
+    Autocomplete,
+    Chip,
+    List,
+    ListItem,
+    MenuItem,
+    Select,
+    Switch,
+    TextField,
+    Tooltip,
+    Typography,
+} from '@mui/material';
+import { useIntl } from 'react-intl';
+
+const useStyles = makeStyles((theme) => ({
+    paramList: {
+        width: '100%',
+        margin: 0,
+    },
+    paramListItem: {
+        justifyContent: 'space-between',
+        gap: theme.spacing(2),
+        paddingLeft: 0,
+        paddingRight: 0,
+    },
+    paramName: {
+        fontWeight: 'bold',
+        minWidth: '30%',
+        overflowWrap: 'anywhere',
+    },
+}));
+
+const FloatRE = /^-?\d*[.,]?\d*([eE]-?\d*)?$/;
+const IntegerRE = /^-?\d*$/;
+const ListRE = /^\[(.*)]$/;
+const sepRE = /[, ]/;
+
+export function extractDefault(paramDescription) {
+    const d = paramDescription.defaultValue;
+    if (paramDescription.type === 'BOOLEAN') return !!d;
+    if (paramDescription.type === 'DOUBLE') return d - 0.0;
+    if (paramDescription.type === 'INTEGER') return d - 0;
+    if (paramDescription.type === 'STRING_LIST') {
+        if (Array.isArray(d)) return d;
+        const mo = ListRE.exec(d);
+        if (mo?.length > 1) {
+            return mo[1]
+                .split(sepRE)
+                .map((s) => s.trim())
+                .filter((s) => !!s);
+        }
+        return [];
+    }
+    return d ?? null;
+}
+
+function longestCommonPrefix(strs) {
+    if (!strs?.length) return '';
+    let prefix = strs.reduce((acc, str) =>
+        str.length < acc.length ? str : acc
+    );
+
+    for (let str of strs) {
+        while (str.slice(0, prefix.length) !== prefix) {
+            prefix = prefix.slice(0, -1);
+        }
+    }
+    return prefix;
+}
+
+/**
+ * Present a "list" of independently editable parameters according to
+ * their description, as given by paramsAsArray, with current values as in initValues.
+ * @param paramsAsArray [{type,name,possibleValues,defaultValue}]
+ * @param initValues {k:v}
+ * @param onChange (paramName, newValue, isInEdition)
+ * @returns JSX
+ */
+export const FlatParameters = ({ paramsAsArray, initValues, onChange }) => {
+    const classes = useStyles();
+    const intl = useIntl();
+
+    const longestPrefix = longestCommonPrefix(paramsAsArray.map((m) => m.name));
+    const lastDotIndex = longestPrefix.lastIndexOf('.');
+    const prefix = longestPrefix.slice(0, lastDotIndex + 1);
+
+    const [uncommitted, setUncommitted] = useState(null);
+    const [inEditionParam, setInEditionParam] = useState(null);
+
+    const preparePossibleValues = useCallback(
+        (values) => {
+            if (values == null) return [];
+            return values
+                .map((v) => intl.formatMessage({ id: v, defaultMessage: v }))
```

Comment:
Should we do the translation here ? 
Or should we give the translated values to the component ?
We have plenty of Error: [@formatjs/intl Error MISSING_TRANSLATION] in the ExportDialog for example.

Reply:
I just tried to report the previous development from Sylvain, when I merged main in my branch.

Reply:
I think we should translate as later as possible (at rendering <FormattedMessage>) when we could.
If we have plenty of missing translation where we need a text to be translated, we must add translation that's all.